gammapy/time/lightcurve.py

```diff
@@ -274,14 +274,50 @@ def _get_off_evt_list(spec_extract):
     @staticmethod
     def _get_on_evt_list(spec_extract):
         """
-        Returns list of OFF events for each observations
+        Returns list of ON events for each observations
         """
         on_evt_list = []
         for obs in spec_extract.bkg_estimate:
             on_evt_list.append(obs.on_events)
 
         return on_evt_list
 
+    def create_fixed_time_bin_lc(self, time_step, spectral_model, energy_range, spectrum_extraction):
+        """
+        Create time intervals of fixed size and then a light curve using those intervals
```

Comment:
Is it possible to only have a utility function to compute the time binning, and then re-use the existing methods?

We're trying to bring Gammapy towards more modular code, and one key to modularity is that one function has one responsibility. Your method has two, as demonstrated by the "and" in the docstring:
> Create time intervals of fixed size and then a light curve using those intervals

We don't know how to organise the LC estimation code yet. For me, it would be reasonable to just accumulate a bit of functionality either as methods on this class, or better yet, as separate standalone functions in the same file.